packages/components/src/edit-in-place-control/style.scss

```diff
@@ -0,0 +1,16 @@
+.components-edit-in-place-control__label {
+	padding: $grid-unit-30;
+	width: auto;
+}
+.components-edit-in-place-control__input {
+	margin-bottom: 1px;
+	text-indent: $grid-unit;
+	&.large {
+		width: 100%;
+		transition: width 0.3s;
+	}
+	&.small {
+		width: 50%;
+		transition: width 0.3s;
```

Comment:
I think github had a glitch as my original review comment disappeared. My comment mentioned that the animation doesn't work as expected when the menu name is long, it becomes smaller and then expands:
![longmenuname](https://user-images.githubusercontent.com/677833/107597734-7e7c6180-6c56-11eb-84f3-00a536b5f641.gif)

I'd expect it to go smoothly from one size to the other.

Reply:
The problem is that the component changes from being a button, that automatically takes the width of its contents, to being an input, that by default takes a width of 20 characters (that can be changed by using the [size attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/text#physical_input_element_size)). So what's happening is we're transitioning to an input with a smaller width than the button, and then because the input's initial size is even smaller than its final size, we see that jump.

There are two ways to approach this: either we set the input `size` to equal `value` and it'll grow or shrink when edited, and always be exactly the same size as the button, or we set a fixed width for both button and input, but then I'm not sure there's any point in making it bigger when it's editable 🤔 

Also: we don't need the `&.small` rule (or the small classname at all); `transition` and the initial `width` value can be set directly on `.components-edit-in-place-control__input`, and then we won't need to repeat `transition` in `&.large` either.

Reply:
I have moved this component to a separate PR https://github.com/WordPress/gutenberg/pull/28924.
In my opinion fixed width of a button and input could be problematic for users when they have longer menu names.
@tellthemachines in order to make an editable area that stores users input and keep the size of text we could either add some js logic to enlarge input while typing or to use span with contenteditable property set to true:

```
<span
	role="textbox"
	aria-label={ editNameInputLabel }
	ref={ inputRef }
	className={ inputCssClasses }
	contentEditable={ true }
	onInput={ ( event ) => {
		const _value = inputRef.current.innerText;
		setValue( _value );
		onChange( _value, event );
	} }
	dangerouslySetInnerHTML={ { __html: value } }
	onFocus={ () =>
		setInputCssClasses( withTransitionCssClass )
	}
	tabIndex="0"
	onBlur={ () => {
		const _value = inputRef.current.innerText;
		setValue( _value );
		setInputCssClasses( withoutTransitionCssClass );
		setIsEdit( false );
	} }
	onKeyDown={ ( event ) => {
		if ( 'Enter' === event.key ) {
			cancelEvent( event );
			if ( inputValidator( value ) ) {
				setIsEdit( false );
				setValue( value );
				onUpdate( value );
			}
		}
		if ( 'Escape' === event.key ) {
			cancelEvent( event );
			setValue( initialValue );
			setIsEdit( false );
			event.target.blur();
		}
		onKeyDown( event );
	} }
/>
```
The input element is more stable, it is easier to add transitions and work with it in general. On the other hand, contenteditable makes the size adjusting trivial.

Reply:
We don't need JS logic to adjust the input; its [size attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/text#physical_input_element_size) does that automatically. Try setting `size` to `value` and you'll see what I mean 🙂

Reply:
Nice idea @tellthemachines :)
Here `value` represents the variable that is displayed inside an input, so the valid expression to pass to `size` attribute would be `String(value).length`. Nevertheless, somehow it is not adjusting properly, and not all chars are visible
![image](https://user-images.githubusercontent.com/8398557/107996115-e5e03b80-6fdf-11eb-9f3b-31660896b4bf.png) 
😕

Reply:
Yeah, `String(value).length.`, of course 😅 
Any CSS attached to the element will override `size`, so you'll have to remove the widths and transitions for it to work.

Reply:
surely I did;) I will make this component available for testing so see what I am talking about.